web/static/js/components/room.jsx

```diff
@@ -1,20 +1,54 @@
-import React from "react"
+import React, { PropTypes } from "react"
 
 import UserList from "./user_list"
 import IdeaBoard from "./idea_board"
 import IdeaGenerationLowerThird from "./idea_generation_lower_third"
+import StageProgressionButton from "./stage_progression_button"
+import PrimeDirective from "./prime_directive"
+import stageProgressionConfigs from "../configs/stage_progression_configs"
 
 import DoorChime from "./door_chime"
 
+import * as AppPropTypes from "../prop_types"
 import styles from "./css_modules/room.css"
 
-const Room = props => (
-  <section className={styles.wrapper}>
-    <IdeaBoard {...props} />
-    <UserList {...props} />
-    <IdeaGenerationLowerThird {...props} />
-    <DoorChime {...props} />
-  </section>
-)
+const Room = props => {
+  if (props.stage === "prime-directive") {
+    const isFacilitator = props.currentUser.is_facilitator
+    const progressionConfig = stageProgressionConfigs[props.stage]
+    return (
+      <section className={styles.wrapper}>
```

Comment:
hmmm.... line 20 and 26 are identical. at the very least, we should avoid repeating that element. going a step further, it'd be nice to wrap the components for the prime directive stage in a `PrimeDirectiveStage` component, that simply wraps the `PrimeDirective`, the `UserList`, and (conditionally) the `StageProgressionButton`, while the any other stage would result in the render of the components in line 27-39. maybe i'm reaching, but it'd be nice to encapsulate the components needed in a particular stage in a particular component.